### PR TITLE
Graph DAO with tests

### PR DESCRIPTION
**Pull request by mattsooknah**
First stab at a graph DAO using Tinkerpop and OrientDB. Also cleaned up the dataaccess package. 

Does not include: using remote DB server, smart transactions.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+import com.tinkerpop.blueprints.{Edge, Vertex, Graph}
+import com.tinkerpop.gremlin.java.GremlinPipeline
+import org.broadinstitute.dsde.rawls.model._
+
+import scala.collection.JavaConversions._
+import scala.collection.JavaConverters._
+
+class GraphEntityDAO(backingDatabase: Graph) extends EntityDAO with GraphDAO {
+  val db = backingDatabase
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala]: why do we have this val if it is the same as the constructor param?

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    val workspaceVertex = verticesRemaining.headOption
+    if (verticesRemaining.count() > 0) {
+      throw new IllegalStateException("Found more than one workspace at " + namespace + "::" + name)
+    }
+    workspaceVertex
+  }
+
+  def getEntityVertex(workspaceNamespace: String, workspaceName: String, entityType: String, entityName: String): Option[Vertex] = {
+    def entityFunc = hasProperty("_name", entityName)
+    getWorkspaceVertex(workspaceNamespace, workspaceName).flatMap(w => {
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: unless my understanding is wrong, this is doing 2 db hits, one to get the workspace then one to get the entity. Can we do this in 1?

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    override def compute(v: Vertex) = props.map(p => v.getProperty(p._1).equals(p._2)).reduce(_&&_)
+  }
+
+  implicit def toPipeFunction[A, B](f: A => B) = new PipeFunction[A, B] {
+    override def compute(a: A): B = f(a)
+  }
+
+  def isReservedProperty(key: String) = key.startsWith("_")
+
+  def getWorkspaceVertex(namespace: String, name: String): Option[Vertex] = {
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphDAO.scala]: it does not seem right to have getWorkspaceVertex and getEntityVertex here. Perhaps instead we have a catalog of filter functions... essentially named queries

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    sourceEntity.addEdge(label, targetVertex)
+  }
+
+  def get(workspaceNamespace: String, workspaceName: String, entityType: String, entityName: String): Option[Entity] = {
+    getEntityVertex(workspaceNamespace, workspaceName, entityType, entityName).map(loadEntity(_, workspaceNamespace, workspaceName))
+  }
+
+  def save(workspaceNamespace: String, workspaceName: String, entity: Entity): Entity = {
+    val entityName = entity.name
+    val entityType = entity.entityType
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphEntityDAO.scala]: why have these vals?

**Comment by mattsooknah** [comment]
I've addressed the comments. It's a bit more elegant now (e.g. passing pipelines around instead of vertices, so we only hit the DB once). Also I added proper attribute serialization (i.e. store properties as their actual Java types instead of Object).

**Pull request merged**